query.go

```diff
@@ -201,14 +204,13 @@ func (r *dhtQueryRunner) spawnWorkers(proc process.Process) {
 			return
 
 		case <-r.rateLimit:
+			ch := r.peersDialed.Consume()
 			select {
-			case p, more := <-r.peersToQuery.DeqChan:
-				if !more {
-					// Put this back so we can finish any outstanding queries.
-					r.rateLimit <- struct{}{}
-					return // channel closed.
+			case p, _ := <-ch:
+				if p == "" {
```

Comment:
It's slightly cleaner to read `p, ok := <-ch` and then check `ok`.

Reply:
👍 For some reason I thought we received the peer and the closure at the same time, but they are different receives.